cmake/nanobind-config.cmake

```diff
@@ -202,8 +202,20 @@ function (nanobind_build_library TARGET_NAME)
   target_compile_definitions(${TARGET_NAME} PRIVATE
     $<${NB_OPT_SIZE}:NB_COMPACT_ASSERTIONS>)
 
-  target_include_directories(${TARGET_NAME} PRIVATE
-    ${NB_DIR}/ext/robin_map/include)
+  # If nanobind was installed without submodule dependencies, then the
+  # dependencies directory won't exist and we need to find them.
+  # However, if the directory _does_ exist, then the user is free to choose
+  # whether nanobind uses them (based on `NB_USE_SUBMODULE_DEPS`), with a
+  # preference to choose them if `NB_USE_SUBMODULE_DEPS` is not defined
+  if (NOT IS_DIRECTORY ${NB_DIR}/ext/robin_map/include OR
+      (DEFINED NB_USE_SUBMODULE_DEPS AND NOT NB_USE_SUBMODULE_DEPS))
+    include(CMakeFindDependencyMacro)
+    find_dependency(tsl-robin-map)
```

Comment:
The `tsl-robin-map` package and `tsl::robin_map` target show up. This, I presume, is an alternative package you have created in the vcpkg ecosystem?

I'm wondering here if there is a way to make this more general. Perhaps, instead of `NB_USE_SUBMODULE_DEPS `, we can let the user specify a target name? (e.g. `NB_TSL_ROBIN_MAP`)

Reply:
The [installation directions](https://github.com/Tessil/robin-map#installation) for robin-map indicate that CMake users can find it in their project with `find_package(tsl-robin-map)`, after it has been installed, and then link against the exported `tsl::robin_map` target. vcpkg has packaged this project, but so have many [other projects](https://repology.org/project/robin-map).

> I'm wondering here if there is a way to make this more general. Perhaps, instead of `NB_USE_SUBMODULE_DEPS`, we can let the user specify a target name? (e.g. `NB_TSL_ROBIN_MAP`)

Sorry, I don't follow what you mean by this. What would this code look like instead?

Reply:
Random comment from a lurker: Conan's tsl-robin-map also uses this name for find_package and its target:
```cmake
find_package(tsl-robin-map REQUIRED CONFIG)
target_link_libraries(${PROJECT_NAME} PRIVATE tsl::robin_map)
```

Reply:
Ok, I was just confused. I did not realize that `tsl::robin_map` has an official CMake installation mechanism. I had been wondering: why is *this* the right target name to use, but it makes sense now.